src/gsym/resolver.rs

```diff
@@ -23,32 +22,55 @@ use super::parser::parse_line_table_header;
 use super::parser::GsymContext;
 use super::types::InfoTypeLineTableInfo;
 
+
+enum MaybeMmap<'dat> {
```

Comment:
given only mmap has file_name associated, I wonder if it makes sense to combine Mmap with filename here?

As for "MaybeMmap" name, something less mmap-centric might make sense? DataSrc or something?

Reply:
> given only mmap has file_name associated, I wonder if it makes sense to combine Mmap with filename here?

I didn't want to do that, because while that condition is true currently, tomorrow we could pass in a "file name for display purposes" along with the raw data. So the two are kind of orthogonal from that perspective.

> As for "MaybeMmap" name, something less mmap-centric might make sense? DataSrc or something?

Sure, will rename.